portal/src/util/misc.ts

```diff
@@ -15,3 +15,24 @@ export function setFieldIfChanged(
     map[field] = value;
   }
 }
+
+export function clearEmptyObject(obj: Record<string, any>): void {
```

Comment:
Can we add unit tests for this function? This should be relatively easier to test.